tests/basic/determinism_check.rs

```diff
@@ -0,0 +1,94 @@
+// use crate::check_replay_roundtrip;
+use shuttle::rand::{thread_rng, Rng};
+// use shuttle::scheduler::RandomScheduler;
+use shuttle::sync::{Mutex, Arc};
+use shuttle::{check_determinism, thread};
+use test_log::test;
+
+
+#[test]
+#[should_panic]
+fn shuttle_determinism_test1() {
```

Comment:
Can we give these tests slightly more suggestive names? This one fails because it randomly acquires a lock, the next one fails because it spawns different numbers of threads.